### PR TITLE
clinical trial db and tasks

### PR DESCRIPTION
**Pull request by kexinhuang12345**

### CODE REVIEW

File: relbench/metrics.py [hunk 1]
```diff
@@ -80,3 +80,24 @@ def rmse(true: NDArray[np.float64], pred: NDArray[np.float64]) -> float:
...
+    return skm.recall_score(true, pred, average='micro')
+
+def multilabel_recall_macro(true: NDArray[np.int_], pred: NDArray[np.int_]) -> float:
+    return skm.recall_score(true, pred, average='macro')
+
+def multilabel_precision_micro(true: NDArray[np.int_], pred: NDArray[np.int_]) -> float:
+    return skm.precision_score(true, pred, average='micro')
+
+def multilabel_precision_macro(true: NDArray[np.int_], pred: NDArray[np.int_]) -> float:
+    return skm.precision_score(true, pred, average='macro')
```
Comment (weihua916) [inline relbench/metrics.py:103]: There are a lot of metrics. Internally, we usually use macro/micro-averaged AUROC or AP. Can we leave a TODO to revisit these metrics?

File: relbench/tasks/trial.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+        return Table(
+            df=df,
+            fkey_col_to_pkey_table={
+                self.source_entity_col: self.source_entity_table,
+                self.destination_entity_col: self.destination_entity_table
+            },
+            pkey_col=None,
+            time_col=self.time_col,
+        )
```
Comment (weihua916) [inline relbench/tasks/trial.py:296]: nit: Add line change

File: relbench/tasks/trial.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        return Table(
+            df=df,
+            fkey_col_to_pkey_table={self.entity_col: self.entity_table},
+            pkey_col=None,
+            time_col=self.time_col,
+        )
+
+
+class WithdrawalTask(RelBenchTask):
+    r"""Predict the the set of reasons of withdrawals for each trial"""
```
Comment (weihua916) [inline relbench/tasks/trial.py:127]: Can we make this task description temporal? predict ... for the next XXX days/year

Comment (weihua916) [reply relbench/tasks/trial.py:127]: Same for the other task descriptions.

File: relbench/tasks/trial.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+
+    name = "rel-trial-withdrawal"
+    task_type = TaskType.MULTILABEL_CLASSIFICATION
+    entity_col = "nct_id"
+    entity_table = "studies"
+    time_col = "timestamp"
+    target_col = "withdraw_reasons"
+    timedelta = pd.Timedelta(days=365)
+    metrics = [mae, rmse]
```
Comment (weihua916) [inline relbench/tasks/trial.py:136]: replace this with multilabel metrics.

File: relbench/datasets/trial.py [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+from relbench.data import Database, RelBenchDataset, Table
+from relbench.tasks.trial import OutcomeTask, AdverseEventTask, WithdrawalTask, SiteSuccessTask, SponsorConditionTask
+from relbench.utils import unzip_processor
+
+
+class TrialDataset(RelBenchDataset):
+    name = "rel-trial"
+    # 2 years gap
+    val_timestamp = pd.Timestamp("2019-01-01")
+    test_timestamp = pd.Timestamp("2021-01-01")
```
Comment (weihua916) [inline relbench/datasets/trial.py:16]: If we were to use 2 years gap here, our timedelta should not be larger than 2 years.

File: relbench/tasks/trial.py [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+class SiteSuccessTask(RelBenchTask):
+    r"""Predict the success rate of a trial site in the next 3 years."""
+
+    name = "rel-trial-site"
+    task_type = TaskType.REGRESSION
+    entity_col = "facility_id"
+    entity_table = "facilities"
+    time_col = "timestamp"
+    target_col = "success_rate"
+    timedelta = pd.Timedelta(days=365*3)
```
Comment (weihua916) [inline relbench/tasks/trial.py:207]: This is 3 years. It may leak information from train to val.

**Review by weihua916 — APPROVED** [review]
Thank you for the effort! LGTM after addressing my comments.

**Review by joshr17 — APPROVED** [review]
Also LGTM! - I added a few points of clarification.

**Comment by kexinhuang12345** [comment]
Thanks for all the comments! I just resolved them and i will merge if there is no other issue!

**Pull request merged**